src/In.ProjectEKA.TMHHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,648 @@
+                                    new JsonSerializerSettings
+                                    {
+                                        Formatting = Formatting.Indented,
+                                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                                    });
+                                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        if (tmhPatientData.DiagnosticReportAsImages != null &&
+                            tmhPatientData.DiagnosticReportAsImages.Any())
+                        {
+                            var diagnosticReportResponse = FetchDiagnosticReportImagesData(dataRequest,
+                                tmhPatientData.DiagnosticReportAsImages, patientName).GetAwaiter().GetResult();
+                            if (diagnosticReportResponse.HasValue)
+                            {
+                                var serializeObject = JsonConvert.SerializeObject(
+                                    diagnosticReportResponse.ValueOr((DiagnosticReportResponse) null),
+                                    new JsonSerializerSettings
+                                    {
+                                        Formatting = Formatting.Indented,
+                                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                                    });
+                                var careBundle = new CareBundle(caseId, parser.Parse<Bundle>(serializeObject));
+                                careBundles.Add(careBundle);
+                            }
+                        }
+
+                        break;
+                    }
+                    case HiType.Medication:
+                        break;
+                    case HiType.DocumentReference:
+                        break;
+                    case HiType.Prescription:
+                        break;
+                    case HiType.DischargeSummary:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            var entries = new Entries(careBundles);
+            return Option.Some(entries);
+        }
+
+        private static async Task<Option<DiagnosticReportResponse>> FetchDiagnosticReportImagesData(
+            DataRequest dataRequest, List<DiagnosticReportAsImage> diagnosticReportAsImages, string patientName)
+        {
+            throw new NotImplementedException();
+        }
+
+        private static async Task<Option<DiagnosticReportResponse>> FetchDiagnosticReportPdfsData(
+            DataRequest dataRequest, List<DiagnosticReportAsPdf> diagnosticReportAsPdfs, string patientName)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'dataRequest'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)